Observer/TruemoneyDataAssignObserver.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace Omise\Payment\Observer;
+
+use Magento\Framework\Event\Observer;
+use Magento\Payment\Observer\AbstractDataAssignObserver;
+use Magento\Quote\Api\Data\PaymentInterface;
+
+class TruemoneyDataAssignObserver extends AbstractDataAssignObserver
```

Comment:
Again, as before we have a lot of repeated code in the `DataAssignObserver`s for a lot of these methods. Three of them (offsite ones?) all feature identical `execute` methods. We should refactor to avoid the repeated code - perhaps inherit from a `OffsiteDataAssignObserver` class?

Reply:
Thanks, This pr implements functionality in current architecture.
Your suggestions will be the subject to future tasks.